dex/src/main/scala/com/wavesplatform/dex/Application.scala

```diff
@@ -457,7 +459,7 @@ class Application(settings: MatcherSettings, config: Config)(implicit val actorS
       matcherQueue.startConsume(firstConsumedOffset, consumeMessages)
     } zip {
       log.info(s"Last queue offset is $lastOffsetQueue")
-      waitOffsetReached(lastOffsetQueue, deadline)
+      WaitOffsetTool.waitOffsetReached(getLastQueueOffset, lastProcessedOffset, lastOffsetQueue, deadline, settings)
```

Comment:
`waitOffsetReached` used `lastProcessedOffset` from `Application` scope, now I put it into another object, so we need to pass it to a new function. 

`startingLastOffset` is really  not nessesary here, but I thought that it would be good, if we got this value before, then why can't we just use it again, if we need to go to kafka for this value? And this value is used for next check:

```
      log.info(s"Offsets: earliest snapshot = $earliestSnapshotOffset, first = $firstQueueOffset, last = $lastOffsetQueue")
      if (lastOffsetQueue < earliestSnapshotOffset)
        Future.failed(RecoveryError("The queue doesn't have messages to recover all orders and continue work. Did you change the queue?"))
      else ....
```

so, we really can ask for this value only in `waitOffsetReached`, but then we need to remove this check or place it in `waitOffsetReached`